### ISSUE TITLE
EFA->EFB Sync: EFA does not safe selected boat types for EFB sync

### ISSUE THREAD

**Issue by cubitusgh2** [nicmichael/efa#50]
Relevant for german canoers only.

EFA can sync trips to EFB (cloud based service from DKV to maintain trips in a canoe club and deliver statistics to organisation).
To do so, you must choose which boat types are relevant for synchronisation.
This is neccessary when for instance, your club does both canoeing and rowing, you only want to sync the canoe trips to the efb.

The list in Configuration -> Synchronisation -> Kanu-EFB is always set to the standard boat types for canoeing.
When you select additional/other boat types and save the settings, the new values are only taken into account until efa boathouse gets restarted (which it does at 4am by default). The selected boat types get not persisted.

This can be fixed by updating EfaConfig.java -> updateValue:
```
 public boolean updateValue(String name, String value) {
        try {
            EfaConfigRecord r = (EfaConfigRecord)data().get(EfaConfigRecord.getKey(name));
            r.setValue(value);
            data().update(r);
            if (r == null) { // this is new. if a value needs an update, but is not available, add it to the config items.
            	addValue(name, value);
            } else {
            	r.setValue(value);
            	data().update(r);
            }
            return true;
        } catch(Exception e) {
            return false;
}  
```

**Comment by cubitusgh2** [comment]
Solution in refactored EFB Sync in https://github.com/cubitusgh2/efa_ksc_replay/tree/EFA_033_EFA-EFB_SYNC_UPDATE

**Comment by cubitusgh2** [comment]
Issue solved with https://github.com/nicmichael/efa/pull/79

**Comment by cubitusgh2** [comment]
Fixed with #79
